lib/ansible/modules/cloud/azure/azure_rm_sqlserver.py

```diff
@@ -0,0 +1,321 @@
+            elif key == "admin_username":
+                self.parameters["administrator_login"] = kwargs[key]
+            elif key == "admin_password":
+                self.parameters["administrator_login_password"] = kwargs[key]
+            elif key == "version":
+                self.parameters["version"] = kwargs[key]
+            elif key == "identity":
+                self.parameters['identity']["type"] = kwargs[key]
+
+        old_response = None
+        results = dict()
+
+        self.mgmt_client = self.get_mgmt_svc_client(SqlManagementClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        try:
+            resource_group = self.get_resource_group(self.resource_group)
+        except CloudError:
+            self.fail('resource group {0} not found'.format(self.resource_group))
+
+        if not ("location" in self.parameters):
+            self.parameters["location"] = resource_group.location
+
+        old_response = self.get_sqlserver()
+
+        if not old_response:
+            self.log("SQL Server instance doesn't exist")
+            if self.state == 'absent':
+                self.log("Old instance didn't exist")
+            else:
+                self.to_do = Actions.Create
+        else:
+            self.log("SQL Server instance already exists")
+            if self.state == 'absent':
+                self.to_do = Actions.Delete
+            elif self.state == 'present':
+                self.log("Need to check if SQL Server instance has to be deleted or may be updated")
+                self.to_do = Actions.Update
+
+        if (self.to_do == Actions.Create) or (self.to_do == Actions.Update):
+            self.log("Need to Create / Update the SQL Server instance")
+
+            if self.check_mode:
+                return self.results
+
+            response = self.create_update_sqlserver()
+            if not old_response:
+                self.results['changed'] = True
+            else:
+                self.results['changed'] = old_response.__ne__(response)
+            self.results.update(response)
+
+            # remove unnecessary fields from return state
+            self.results.pop('name', None)
+            self.results.pop('type', None)
+            self.results.pop('tags', None)
+            self.results.pop('location', None)
+            self.results.pop('identity', None)
+            self.results.pop('kind', None)
+            self.results.pop('administrator_login', None)
+            self.results.pop('administrator_login_password', None)
+            self.log("Creation / Update done")
+        elif self.to_do == Actions.Delete:
+            self.log("SQL Server instance deleted")
+            self.delete_sqlserver()
+            self.results['changed'] = True
+        else:
+            self.log("SQL Server instance unchanged")
+            self.results['state'] = old_response
```

Comment:
This shouldn't be returned, it should be the same values that you set in add or update.